### PR TITLE
Task criar metodo para consulta no codex neodv 1680

### PR DESCRIPTION
**Pull request by henrique-lh**
## Description

With this pr:
- Add describe for training v2
- Add modes for describing

## Related issues

- Closed: https://linear.app/datarisk/issue/NEODV-1680/[task]-criar-metodo-para-consulta-no-codex

## How to test it

- Setup your environment
- Test the `notebooks/TrainingV2.ipynb`
 
 
 
 
 
 **PR Summary by Typo**
------------

**Overview**
This PR modifies the `base.py` and `training.py` files within the `mlops_codex` library. It removes the promotion functionality from the base training class and introduces a `project_info` method to retrieve project details. Additionally, it adds a `list` method to the `MLOpsTrainingClient` to list available trainings.

**Key Changes**
- Removed the `__promote` and `promote` methods from `base.py`, raising a `NotImplementedError`.
- Replaced the `__describe` endpoint URL in `training.py` and renamed the `executions` method to `project_info`, adding a 'log' mode to output YAML data.
- Added a `list` method to `MLOpsTrainingClient` to retrieve and display training data in different formats ('dict', 'count', 'log').
- Updated docstrings and examples to reflect the changes.

**Recommendations**
Not deployment ready. While the changes seem reasonable, the removal of promotion functionality and the addition of new methods require thorough testing to ensure stability and avoid unintended consequences. Add unit tests for the new `project_info` and `list` methods, covering different modes and edge cases.  Also, provide clear documentation explaining the removal of the promotion feature and the intended usage of the new methods.
 

 <h6>To turn off PR summary, please visit <a href="https://app.typoapp.io/settings/notification?tab=codeHealth">Notification settings</a>.</h6>

### CODE REVIEW

File: src/mlops_codex/training/training.py [hunk 1]
```diff
@@ -498,7 +499,7 @@ def __describe(self):
         AuthenticationError 
             Invalid credentials
         """
-        url = f"{self.base_url}/training/describe/{self.group}/{self.training_hash}"
+        url = f"{self.base_url}/v2/training/{self.training_hash}"
```
[the remark this answers is not in the document]
Comment (64J0) [reply src/mlops_codex/training/training.py:502]: It seems enough to have it in a single place, thanks

**Review by 64J0 — APPROVED** [review]

File: src/mlops_codex/training/training.py [hunk 2]
```diff
@@ -577,11 +572,14 @@ def executions(self, mode="dict"):
...
         """
-        describe = self.__describe().get("Executions")
+        describe = self.__describe()
         if mode == "dict":
             return describe
         elif mode == "count":
-            return len(describe)
+            return describe["ExperimentsQuantity"]
+        elif mode == 'log':
+            yaml_data = parse_json_to_yaml(describe)
```
Comment (johnrc90) [inline src/mlops_codex/training/training.py:581]: A informação das execuções vai vir dos novos endpoints [daqui](https://linear.app/datarisk/issue/NEODV-1639/[task]-adicionar-novos-endpoints-de-consulta-de-execucao), então esse metodo não retorna nada das execuções além do total. Seria melhor usar isso do describe num metodo tipo o `project_info`

Comment (henrique-lh) [reply src/mlops_codex/training/training.py:565]: https://github.com/datarisk-io/mlops_codex/pull/78/commits/87b328fb80f021717ce929ece7269928f9f7da03

Since this pr will be merged before the copy pr, I'll update the training notebook in the copy pr
